tensorflow/tools/ci_build/linux/rocm/run_py3_core.sh

```diff
@@ -19,21 +19,81 @@ set -e
 set -x
 
 N_JOBS=$(grep -c ^processor /proc/cpuinfo)
+N_GPUS=$(lspci|grep 'VGA'|grep 'AMD/ATI'|wc -l)
 
 echo ""
-echo "Bazel will use ${N_JOBS} concurrent job(s)."
+echo "Bazel will use ${N_JOBS} concurrent build job(s) and ${N_GPUS} concurrent test job(s)."
 echo ""
 
 # Run configure.
 export PYTHON_BIN_PATH=`which python3`
 export CC_OPT_FLAGS='-mavx'
 
 export TF_NEED_ROCM=1
+export TF_GPU_COUNT=${N_GPUS}
 
 yes "" | $PYTHON_BIN_PATH configure.py
 
 # Run bazel test command. Double test timeouts to avoid flakes.
-bazel test --config=rocm --test_tag_filters=-no_oss,-oss_serial,-no_gpu,-benchmark-test -k \
-    --test_lang_filters=py --jobs=${N_JOBS} --test_timeout 300,450,1200,3600 \
-    --build_tests_only --test_output=errors --local_test_jobs=1 --config=opt \
-    //tensorflow/... -//tensorflow/compiler/... -//tensorflow/contrib/...
+bazel test --test_sharding_strategy=disabled --config=rocm --test_tag_filters=-no_oss,-oss_serial,-no_gpu,-benchmark-test -k \
+    --test_lang_filters=py --jobs=${N_JOBS} --test_timeout 600,900,2400,7200 \
+    --build_tests_only --test_output=errors --local_test_jobs=${TF_GPU_COUNT} --config=opt \
+    --run_under=//tensorflow/tools/ci_build/gpu_build:parallel_gpu_execute -- \
+    //tensorflow/... -//tensorflow/compiler/... -//tensorflow/contrib/... \
+    -//tensorflow/python/distribute:distribute_coordinator_test \
```

Comment:
Instead of a hardcoded list, can we add "no_rocm" or "no_amd" tag and just add that tag to the test_tag_filters flag here?

Reply:
Thanks for the review.  Sounds like a cleaner strategy.  We'll give that a try.

Reply:
If this script already produces broken builds, Let's just change this file to add the tag filter here, then we can tag the targets gradually, piece by piece.
WDYT?

Reply:
Good idea, that sounds reasonable.  

i.e. Use this PR, remove all of the extra excluded tests, but add the `-no_rocm` tag.  Then, create a follow-on PR that'll add those tags to each of the individual tests.

Reply:
Does that look better now?